code/modules/mob/living/simple_animal/hostile/terror_spiders/objects.dm

```diff
@@ -0,0 +1,353 @@
+
+#define SPINNING_WEB 1
+#define LAYING_EGGS 2
+#define MOVING_TO_TARGET 3
+#define SPINNING_COCOON 4
+
+// --------------------------------------------------------------------------------
+// ----------------- TERROR SPIDERS: EGGS (USED BY NURSE AND QUEEN TYPES) ---------
+// --------------------------------------------------------------------------------
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/DoLayTerrorEggs(var/lay_type, var/lay_number, var/lay_crawl)
+	stop_automated_movement = 1
+	var/obj/effect/spider/terror_eggcluster/C = new /obj/effect/spider/terror_eggcluster(get_turf(src))
+	C.spiderling_type = lay_type
+	C.spiderling_number = lay_number
+	C.spiderling_ventcrawl = lay_crawl
+	C.faction = faction
+	C.spider_myqueen = spider_myqueen
+	C.master_commander = master_commander
+	C.ai_playercontrol_allowingeneral = ai_playercontrol_allowingeneral
+	C.enemies = enemies
+	if (lay_type == /mob/living/simple_animal/hostile/poison/terror_spider/red)
```

Comment:
you should define
`var/egg_name` on `/mob/living/simple_animal/hostile/poison/terror_spider/`
and set it on each of these paths
then use `C.name = initial(lay_type.egg_name)`